overlord/snapstate/handlers.go

```diff
@@ -4117,10 +4125,30 @@ func (m *SnapManager) doEnforceValidationSets(t *state.Task, _ *tomb.Tomb) error
 		return err
 	}
 
-	if err := EnforceValidationSets(st, decodedAsserts, pinnedSeqs, snaps, ignoreValidation, userID); err != nil {
-		return fmt.Errorf("cannot enforce validation sets: %v", err)
-	}
+	if local {
+		vsKeys := make(map[string][]string)
+		if err := t.Get("validation-set-keys", &vsKeys); err != nil {
```

Comment:
in theory given this we might not need to introduce "local" but simply check for this vs validation-sets